TACC/Core-Portal
# task/WP-271 Upgrade Django and 3rd party dependancies

## Overview

Updates Django to version 4.2, addresses deprecation warnings and any breaking changes. 

Resources: 

- [4.0 Release Notes](https://docs.djangoproject.com/en/4.2/releases/4.0/)
- [4.1 Release Notes](https://docs.djangoproject.com/en/4.2/releases/4.1/)
- [4.2 Release Notes](https://docs.djangoproject.com/en/4.2/releases/4.2/)

## Related

* [WP-271](https://jira.tacc.utexas.edu/browse/WP-271)

## Changes

- Updated django version and other dependancies that were throwing deprecation errors 

## Testing

1. Go to `/server` and run `poetry install `
2. Run `make build`
3. Run migrations 
4. Ensure everything works as expected, monitor docker logs for any errors thrown
5. Run server testing and make sure no warning messages are shown

## UI



## Notes

Added comments for some stuff I am not sure about so would appreciate some feedback on that
